tests/test_core.py

```diff
@@ -783,6 +783,13 @@ def __init__(self):
         imageio.imwrite("foo.bmp", "asd")
 
 
+def test_imwrite_symbol_name():
+    name = '''#!~@$%^&`-+{};',.() []_=.jpg'''
+    imageio.imwrite(name, np.zeros((128, 128, 3), dtype=np.uint8))
+    im = imageio.imread(name)
+    assert im.shape == (128, 128, 3)
```

Comment:
```suggestion
    tmp_request = imageio.core.Request(name, "r")
    assert tmp_request.extension == ".jpg"
    tmp_request.finish()
```

I think explicitly testing the `Request` object here is cleaner. This way we don't need a plugin installed that can read/write JPG. It also tests that the extension is what we expect it to be, which was the primary symptom that made us notice this.

Alternatively, we can do both assertions. Pillow is a core dependency, so we should always be capable of reading and writing JPEG.